esphome/components/i2s_audio/speaker/i2s_audio_speaker.cpp

```diff
@@ -40,30 +40,47 @@ void I2SAudioSpeaker::player_task(void *params) {
   xQueueSend(this_speaker->event_queue_, &event, portMAX_DELAY);
 
   i2s_driver_config_t config = {
-      .mode = (i2s_mode_t) (I2S_MODE_MASTER | I2S_MODE_TX),
-      .sample_rate = 16000,
-      .bits_per_sample = I2S_BITS_PER_SAMPLE_16BIT,
-      .channel_format = I2S_CHANNEL_FMT_RIGHT_LEFT,
-      .communication_format = I2S_COMM_FORMAT_STAND_I2S,
-      .intr_alloc_flags = ESP_INTR_FLAG_LEVEL1,
-      .dma_buf_count = 8,
-      .dma_buf_len = 128,
-      .use_apll = false,
-      .tx_desc_auto_clear = true,
-      .fixed_mclk = I2S_PIN_NO_CHANGE,
-      .mclk_multiple = I2S_MCLK_MULTIPLE_DEFAULT,
-      .bits_per_chan = I2S_BITS_PER_CHAN_DEFAULT,
+    .mode = (i2s_mode_t) (this_speaker->parent_->get_i2s_mode() | I2S_MODE_TX),
+    .sample_rate = 16000,
+    .bits_per_sample = this_speaker->bits_per_sample_,
+    .channel_format = I2S_CHANNEL_FMT_RIGHT_LEFT,
```

Comment:
```suggestion
    .channel_format = I2S_CHANNEL_FMT_ONLY_RIGHT,
```
If you set the channel_format here there is no need of calling i2s_set_clk

Reply:
Thanks!